wrapanapi/systems/nuage.py

```diff
@@ -58,26 +70,191 @@ def common_headers(self):
             'X-Nuage-Organization': 'csp'
         }
 
+    #
+    # Integration-test specific entities on Nuage server
+    #
+
+    @property
+    def enterprise(self):
+        if not self._enterprise:
+            match = self.list_enterprises(name=self.enterprise_name)
+            if match:
+                self._enterprise = match[0]['ID']
```

Comment:
This should be logged when there was no match for the instance's set enterprise_name within the listed enterprises, instead of it just always being None.

This pattern is also creating a bit of an odd pattern in that `self._enterprise' and `self.enterprise` are actually an enterprise ID and not the enterprise object itself.  Consider setting it to the actual enterprise object, or modify the names.

Seems like this could be done within the constructor, since there's no setter for the property and once its set it won't change via calling the property.